common/client/versionChecker.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-version"
+	"go.uber.org/yarpc"
+
+	"github.com/uber/cadence/.gen/go/shared"
+	"github.com/uber/cadence/common"
+)
+
+const (
+	// GoSDK is the header value for common.ClientImplHeaderName indicating a go sdk client
+	GoSDK = "uber-go"
+	// JavaSDK is the header value for common.ClientImplHeaderName indicating a java sdk client
+	JavaSDK = "uber-java"
+	// CLI is the header value for common.ClientImplHeaderName indicating a cli client
+	CLI = "cli"
+
+	// SupportedGoSDKVersion indicates the highest go sdk version server will accept requests from
+	SupportedGoSDKVersion = "1.4.0"
+	// SupportedJavaSDKVersion indicates the highest java sdk version server will accept requests from
+	SupportedJavaSDKVersion = "1.4.0"
+	// SupportedCLIVersion indicates the highest cli version server will accept requests from
+	SupportedCLIVersion = "1.4.0"
+
+	// GoWorkerStickyQueryVersion indicates the minimum client version of go worker which supports StickyQuery
+	GoWorkerStickyQueryVersion = "1.0.0"
+	// JavaWorkerStickyQueryVersion indicates the minimum client version of the java worker which supports StickyQuery
+	JavaWorkerStickyQueryVersion = "1.0.0"
+	// GoWorkerConsistentQueryVersion indicates the minimum client version of the go worker which supports ConsistentQuery
+	GoWorkerConsistentQueryVersion = "1.5.0"
+
+	stickyQuery     = "sticky-query"
+	consistentQuery = "consistent-query"
+)
+
+type (
+	// VersionChecker is used to check client/server compatibility and client's capabilities
+	VersionChecker interface {
+		ClientSupported(context.Context, bool) error
```

Comment:
In general, can you use named params in the interface method so people know what these variables mean? (e.g. this bool...)